lib/cereal/postgres/postgres.go

```diff
@@ -29,6 +29,29 @@ const (
 	failWorkflowQuery     = `SELECT cereal_fail_workflow($1, $2)`
 	continueWorkflowQuery = `SELECT cereal_continue_workflow($1, $2, $3, $4, $5)`
 
+	listWorkflowInstancesQuery = `
+WITH res AS (
+	SELECT DISTINCT ON (instance_name, workflow_name) *
+    FROM cereal_workflow_results
+	ORDER BY instance_name, workflow_name, end_at DESC
+)
+SELECT COALESCE(a.id, res.id) id, 
+       COALESCE(a.status, 'completed') status,
+       COALESCE(a.instance_name, res.instance_name) instance_name,
+	   COALESCE(a.workflow_name, res.workflow_name) workflow_name,
+	   COALESCE(a.parameters, res.parameters) parameters,
+	   a.payload payload,
+	   res.result result,
+	   res.error error
```

Comment:
This is almost a SQL koan or something. Writing a query that joins two tables and optionally supports three different where options! There is a dragon here where we could join two workflows that share the same workflow name and instance name but we match the result of a previous instance to the currently running instance! That's right, a running instance that's already completed!

You handle it when you're scanning through the rows which is dope, but I think we should write a comment about that side effect of this query so future us (me) isn't confused about it.